### PR TITLE
Seclude general error handler and improve process execution span setup

### PR DESCRIPTION
**Pull request by medikoo**
Addresses 0.1 and 0.2 from https://github.com/serverless/serverless/issues/8364

Additionally fixed tests in two unrelated files as they were immune to side effects of already required utils (ensured there `runServerless` runs always run with clear require cache)

### CODE REVIEW

File: lib/cli/handle-error.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  }
+  return {
+    title: 'Exception',
+    message: inspect(exception),
+  };
+};
+
+const consoleLog = (message) => process.stdout.write(`${message}\n`);
+
+const writeMessage = (title, message) => {
```
Comment (pgrzesik) [inline lib/cli/handle-error.js:34]: This is duplicated across two modules or I'm missing the difference between them?

**Review by pgrzesik — CHANGES_REQUESTED** [review]
One comment about duplicated method, otherwise looks great :+1:

File: lib/cli/handle-error.js (hunk 1 above)
Comment (medikoo) [reply lib/cli/handle-error.js:34]: Good point, I didn't notice (linter simply told me it's still used in `Error.js` so I didn't touch it).

As I see this logic is also used in `logWarning`, and it's the reason it's also kept there.

I'm not sure is it worth to clean this up, as (1) `logWarning` and all other log functions will be removed with https://github.com/serverless/serverless/issues/1720 which is also on priority list, and (2) `handle-error.js` as introduced here is also likely to be improved  (in context of this PR we simply take this logic out of `Serverless` class context, so `Serverless` is not mangled with CLI specific logic)

It's likely that quite soon `writeMessage` will be gone from both locations (I also personally don't see it as part of `cli` utils), and investing time now to make a tested reusable util from it now, doesn't feel that justified.

Let me know

Comment (pgrzesik) [reply lib/cli/handle-error.js:34]: If we'll be removing it soon I see zero harm in keeping in duplicated for this short period of time :+1:

**Review by pgrzesik — APPROVED** [review]

**Pull request merged**